### ISSUE TITLE
[DateInput] Selecting 1st of month breaks in dev mode and creates error message in console

### ISSUE THREAD

**Issue by Sascha-Gschwind** [palantir/blueprint#4399]
## Environment

- __Package version(s)__: @blueprintjs/core: 3.35.0, @blueprintjs/datetime: 3.19.3
- __Operating System__: Windows 10
- __Browser name and version__: Chrome Version 86.0.4240.111

#### Code Sandbox
Check out console message on: https://blueprintjs.com/docs/#datetime/dateinput
![image](https://user-images.githubusercontent.com/17166009/98102733-3e0d8a80-1e94-11eb-81ad-28763e56a3b2.png)

#### Steps to reproduce

1. Select 1.11.2020 as a date (or any 1st of month date) and watch the console error

#### Actual behavior

- querySelectorAll Error in console.
- If you are developing the app crashes.

#### Expected behavior

- no error in console.
- If you are developing the app should not crash.

#### Possible solution

Fix [handlePopoverBlur](https://github.com/palantir/blueprint/blob/526722781159da8f2992d26fc2026b062de4cd92/packages/datetime/src/dateInput.tsx#L428) or [getLastTabbableElement](https://github.com/palantir/blueprint/blob/526722781159da8f2992d26fc2026b062de4cd92/packages/datetime/src/dateInput.tsx#L413)

The element which gets selected [here](https://github.com/palantir/blueprint/blob/526722781159da8f2992d26fc2026b062de4cd92/packages/datetime/src/dateInput.tsx#L416) seems to be null and this breaks the application

The else case in the [handlePopoverBlur](https://github.com/palantir/blueprint/blob/526722781159da8f2992d26fc2026b062de4cd92/packages/datetime/src/dateInput.tsx#L428) doesn't take into account that on chrome the popoverElement is sometimes null (as written in the comment [here](https://github.com/palantir/blueprint/blob/526722781159da8f2992d26fc2026b062de4cd92/packages/datetime/src/dateInput.tsx#L435))

I'd probably fix [getLastTabbableElement](https://github.com/palantir/blueprint/blob/526722781159da8f2992d26fc2026b062de4cd92/packages/datetime/src/dateInput.tsx#L413) and make it more robust. It should still do the same but doesn't break if `this.popoverContentEl` is null:

```
    private getLastTabbableElement = () => {
        // Popover contents are well structured, but the selector will need
        // to be updated if more focusable components are added in the future
        const tabbableElements = this.popoverContentEl?.querySelectorAll("input, [tabindex]:not([tabindex='-1'])");
        const numOfElements = tabbableElements?.length;
        // Keep track of the last focusable element in popover and add
        // a blur handler, so that when:
        // * user tabs to the next element, popover closes
        // * focus moves to element within popover, popover stays open
        const lastTabbableElement = numOfElements && numOfElements > 0 ? tabbableElements[numOfElements - 1] : null;

        return lastTabbableElement as HTMLElement;
    };
```

**Comment by iChebbi** [comment]
I had same issue, but it seems that it has been fixed with [`@blueprintjs/datetime 3.20.0`](https://github.com/palantir/blueprint/wiki/3.x-Changelog#blueprintjsdatetime-3200) in #4412

**Comment by Sascha-Gschwind** [comment]
I can confirm it fixes the issue. Thanks for letting me know!
